src/Features/Core/Portable/LegacySolutionEvents/ILegacyWorkspaceDescriptor.cs

```diff
@@ -0,0 +1,18 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using Microsoft.CodeAnalysis.Host;
+
+namespace Microsoft.CodeAnalysis.LegacySolutionEvents
+{
+    internal interface ILegacyWorkspaceDescriptor
+    {
+        /// <inheritdoc cref="Workspace.Kind"/>
+        string? WorkspaceKind { get; }
+        /// <inheritdoc cref="HostWorkspaceServices.SolutionServices"/>
+        SolutionServices SolutionServices { get; }
+        /// <inheritdoc cref="Workspace.CurrentSolution"/>
+        Solution CurrentSolution { get; }
+    }
```

Comment:
this is an abstraction that allows us to just get teh data we need from a workspace, without actually exposing the workspace itself.  

Note that CurrentSolution is oogy as heck, but needed for SolutionCrawler (which commonly goes back to this).  See the docs here though on why this is ok for this legacy/compat purpose: https://github.com/dotnet/roslyn/pull/64150/files#diff-3e98ede73fac6d34d191fb31d946bf9df7eca9d83b24fa662a1c4e8cd30c8e94R68